### PR TITLE
Fix build issues

### PR DESCRIPTION
**Pull request by Indumathi31**
1. Fix build issues
2. Fix core dump issue in the sequence reader.

### CODE REVIEW

File: rocAL/rocAL/source/master_graph.cpp [hunk 1]
```diff
@@ -839,22 +843,26 @@ void MasterGraph::output_routine()
 {
     _process_time.start();
     INFO("Output routine started with "+TOSTR(_remaining_count) + " to load");
+    size_t batch_ratio = _is_sequence_reader_output ? _sequence_batch_ratio : _user_to_internal_batch_ratio;
+    if(!_is_sequence_reader_output) // _sequence_batch_ratio and _user_to_internal_batch_ratio is different. Will be removed in TensorSupport.
```
Comment (rrawther) [inline rocAL/rocAL/source/master_graph.cpp:847]: why is this only done when !_is_sequence_reader_output?

File: rocAL/rocAL/source/master_graph.cpp [hunk 2]
```diff
@@ -1076,14 +1081,23 @@ void MasterGraph::output_routine_video()
...
 void MasterGraph::start_processing()
 {
     _processing = true;
-    _is_video_loader ? (_remaining_count = _video_loader_module->remaining_count(),
-                        _output_thread = std::thread(&MasterGraph::output_routine_video, this))
-                        : (_remaining_count = _loader_module->remaining_count(),
-                        _output_thread = std::thread(&MasterGraph::output_routine, this));
+    if(_is_video_loader)
+    {
+#ifdef RALI_VIDEO
```
Comment (rrawther) [inline rocAL/rocAL/source/master_graph.cpp:1091]: put the #ifdef around the entire block

**Review by rrawther — CHANGES_REQUESTED** [review]

File: rocAL/rocAL/source/master_graph.cpp [hunk 3]
```diff
@@ -458,9 +458,11 @@ MasterGraph::reset()
     // clearing meta ring buffer
     if(_is_video_loader)
     {
+#ifdef RALI_VIDEO
```
[the remark this answers is not in the document]
Comment (rrawther) [reply rocAL/rocAL/source/master_graph.cpp:461]: What I meant was the following
#ifdef RALI_VIDEO
    if(_is_video_loader)
    {
        t = _video_loader_module->timing();
        t.video_process_time += _process_time.get_timing();
    }
    else
#endif

File: rocAL/rocAL/source/master_graph.cpp (hunk 2 above)
Comment (rrawther) [reply rocAL/rocAL/source/master_graph.cpp:1091]: Please see the previous comment

**Comment by rrawther** [comment]
@paveltc : Please run unit-tests on this PR

**Comment by paveltc** [comment]
@rrawther: I will have the unit tests run and give an update by EOD Thursday 11/18/21 PST.

**Comment by Indumathi31** [comment]
@rrawther Have added the change suggested.

**Comment by paveltc** [comment]
@rrawther This PR passes the rali_unittests and the rali_video_unittests for both the OpenCL and HIP backends.

**Review by rrawther — APPROVED** [review]

**Pull request merged**